### PR TITLE
Added Black and pre commit

### PR DESCRIPTION
**Pull request by dcdarrell9**
# Context

I've had a play about with adding some pre-commit hooks in (quality control) and think it might be worth doing, found a pretty good site, thanks to @treilly94 - https://pre-commit.com/
 
# How to review

Check what other types of things we can do with pre-commit hooks and see how it runs on your machine.
Check the formatter to make sure it suits you (We can adapt it how we want)

# Checklist
- [x] Code has been tested
- [ ] Bug/Feature is tested/reviewed against acceptance criteria
- [ ] Bug/Feature passes all tests and passes ci/cd analysis tools
- [x] Code is review ready
- [ ] Bug/Feature is documented (Handover/Knowledge share included)

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -4,12 +4,12 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-flask = "*"
-structlog = "*"
-
-[dev-packages]
```
Comment (treilly94) [inline Pipfile:10]: Are we ok with this removing dev-packages?

Comment (dcdarrell9) [reply Pipfile:10]: I'm happy to have with or without, but just wondering if there's a reason to have them?
I'm guessing the idea is to not have the tests/lints in the actual build, but how much space do they actually take up?
@emilioward Thoughts?

Comment (dcdarrell9) [reply Pipfile:10]: I'm also guessing the idea is to run `pipenv .... --dev` ? So we would need an extra command in the Makefile

Comment (treilly94) [reply Pipfile:10]: Yer, I think the idea is to not include dev dependencies in the proper builld.  
Speaking of the makefile I was thinking we may want a `setup` command that runs  
`pipenv clean`  
`pipenv install --dev`  
`pre-commit install`

could maybe do that stuff in another ticket though

**Pull request merged**